cpp/locality/AABBQuery.cc

```diff
@@ -66,6 +66,13 @@ void AABBIterator::updateImageVectors(float r_max, bool _check_r_max)
     box::Box box = m_neighbor_query->getBox();
     vec3<float> nearest_plane_distance = box.getNearestPlaneDistance();
     vec3<bool> periodic = box.getPeriodic();
+
+    
+        if (r_max <= 0)
+    {
+        throw std::invalid_argument("RDF requires r_max to be positive.");
+    }
```

Comment:
The `r_max` and `r_min` parameters should be validated in the parent class `NeighborQueryPerPointIterator`, not in this method. Validation should have already occurred before reaching this point in the code.

Revert this change and add validation logic to the empty constructor body here instead:
https://github.com/glotzerlab/freud/blob/eb2a97bac302ed45236f26e0a7e6a0346071c579/cpp/locality/NeighborQuery.h#L262-L268

edit: It looks like you have already added validation logic to that method. 👍